app/assets/javascripts/components/autocomplete.js

```diff
@@ -133,5 +135,55 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     $select.parentNode.removeChild($select)
   }
 
+  Autocomplete.prototype.initAutoCompleteSearchTopics = function () {
+    var millerColumns = document.querySelector('miller-columns')
+    var topics = millerColumns.taxonomy.flattenedTopics
+
+    if (!topics) {
+      return
+    }
+
+    new window.accessibleAutocomplete({ // eslint-disable-line no-new, new-cap
+      id: 'topics-autocomplete',
+      name: 'topics-autocomplete',
+      element: this.$module,
+      minLength: 3,
+      autoselect: false,
+      source: function (query, syncResults) {
+        var results = topics
+
+        syncResults(query
+          ? results.filter(function (result) {
+            var indexOf = result.topicName.toLowerCase().indexOf(query.toLowerCase())
+            var resultContainsQuery = indexOf !== -1
+            if (resultContainsQuery) {
+              // Strip existing <mark> tags
+              result.highlightedTopicName = result.topicName.replace(/<\/?mark>/gm, '')
+              // Wrap query in <mark> tags
+              var queryRegex = new RegExp('(' + query + ')', 'ig')
+              result.highlightedTopicName = result.topicName.replace(queryRegex, '<mark>$1</mark>')
+            }
+            return resultContainsQuery
+          }) : []
+        )
+      },
+      templates: {
+        inputValue: function (result) {
+          return result && result.topicName && ''
```

Comment:
What does this do? I'd have thought it always returned '' ?

Reply:
Default to an empty string as an input value (if `result.topicName` doesn't return anything), which enables the autocomplete to be cleared after selection.

Reply:
Not sure if I'm missing something but this will return an empty string even if topicName does return something won't it.

e.g. 

```
"this" && "that" && ""
> ""
```

```
"this" && "that" && ""
> ""

```
"this" && null && ""
> null
```

Reply:
Right, we don't need to check for `result` in our case, only to return the empty string.